### ISSUE TITLE
Compound keys are used incorrectly in writing nodes by batch

### ISSUE THREAD

**Issue by moxious** [neo4j-contrib/neo4j-spark-connector#179]
Wrong cypher is getting generated when using a compound, aliased key for nodes.

Here's my spark code (pay particular attention to `loc_pk`)

```
# (Do complicated stuff to make a "locations" dataframe") 

locations_pk = ['LocationName', 'LocationType', 'FeatureID']
loc_pk = 'LocationName:name,LocationType:type,FeatureID:featureId'

locations.show()

locations.repartition(4).write \
  .format("org.neo4j.spark.DataSource") \
  .mode('overwrite') \
  .option("url", dbutils.widgets.get("url")) \
  .option("authentication.basic.username", dbutils.widgets.get("user")) \
  .option("authentication.basic.password", dbutils.widgets.get("password")) \
  .option("labels", 'Location') \
  .option("node.keys", loc_pk) \
  .option("batch.size", 20000) \
  .save()
```

Here's a picture on what the locations table looks like:
```
+------------------+------------+--------------------+----+--------+--------+---------+--------+---------+----+
|       GKGRECORDID|LocationType|        LocationName|ISO2|ADM1Code|ADM2Code|      Lat|     Lon|FeatureID|Last|
+------------------+------------+--------------------+----+--------+--------+---------+--------+---------+----+
|20200731154500-998|           2|Texas, United States|  US|    USTX|        |   31.106|-97.6475|       TX|4442|
|20200731154500-998|           1|               China|  CH|      CH|        |     35.0|   105.0|       CH|1633|
|20200731154500-998|           1|               China|  CH|      CH|        |     35.0|   105.0|       CH|2401|
|20200731154500-998|           1|               China|  CH|      CH|        |     35.0|   105.0|       CH|4675|
```

While running this and monitoring the Neo4j instance, the Cypher I see generated is this:

```
UNWIND $events AS event
MERGE (node:Location {LocationName: event.keys.LocationName, LocationType: event.keys.LocationType, FeatureID: event.keys.FeatureID})
SET node += event.properties
```

Remember: `loc_pk = 'LocationName:name,LocationType:type,FeatureID:featureId'`

The cypher I would EXPECT is this:

```
UNWIND $events AS event
MERGE (node:Location {name: event.keys.LocationName, type: event.keys.LocationType, featureId: event.keys.FeatureID})
SET node += event.properties
```

Basically -- the merge property list is named incorrectly.

**Comment by conker84** [comment]
closed via 407c5e0
